pyomo/contrib/doe/doe.py

```diff
@@ -1071,14 +1116,21 @@ def fim_rule(m, p, q):
 
     def _add_objective(self, m):
 
-        ### Initialize the Cholesky decomposition matrix
-        if self.Cholesky_option:
+        small_number = 1e-10
+
+        # Assemble the FIM matrix. This is helpful for initialization!
+        fim = np.zeros((len(self.param), len(self.param)))
+        for i, bu in enumerate(m.regression_parameters):
+            for j, un in enumerate(m.regression_parameters):
+                # Copy value from Pyomo model into numpy array
+                fim[i][j] = m.fim[bu, un].value
```

Comment:
It might be more efficient to form the NP array in one shot (from a list or using `fromiter`), and then reshaping to the 2-D matrix

Reply:
I decided to add this as a comment and revisit as we refactor the code over the summer.